src/app/qgsattributesformproperties.h

```diff
@@ -195,6 +194,11 @@ class APP_EXPORT QgsAttributesFormProperties : public QWidget, private Ui_QgsAtt
     void initInitPython();
     void initSuppressCombo();
 
+  public slots:
+
+    void onInvertSelectionButtonClicked( bool checked );
+    void onAttributeSelectionChanged();
```

Comment:
Was that moved here on purpose?
Could it be private slots?

Reply:
What do you mean? I just added slot, they were already public.

Reply:
Question 1 was if `onAttributeSelectionChanged` needs to be a slot (I couldn't find any reference to it in this PR)
Question 2 if the newly added slot needs to be public.

Reply:
> Question 1 was if `onAttributeSelectionChanged` needs to be a slot (I couldn't find any reference to it in this PR)

I don't think it makes any difference whether it's a slot or not ( `slot` keyword it's technically not required in this case), but I think it's clearer to mark it as a slot since it's connected:
 
https://github.com/qgis/QGIS/blob/d317cf4658866da0e2ec1c7dff9e1f5c2b20fa31/src/app/qgsattributesformproperties.cpp#L66

> Question 2 if the newly added slot needs to be public.

No, it does not need to be public, it's just called from within the class, I'll fix that.

I think that the pre-existing `onAttributeSelectionChanged();` also didn't need to be public, it was added by:

8f895f6c645 (David              2017-08-21 08:40:24 +0200)     void onAttributeSelectionChanged();

Reply:
> I don't think it makes any difference whether it's a slot or not ( slot keyword it's technically not required in this case), but I think it's clearer to mark it as a slot since it's connected.

Interesting, it must be used like a lambda in this case, I never noticed that this is possible. But yes, good move to make it a slot too for consistency...
Slots do add some overhead in the moc headers, so I wonder what benefits we still get if connections work without (python bindings maybe, but that's not a concern in app, others?)

> I think that the pre-existing onAttributeSelectionChanged(); also didn't need to be public, it was added by.

Yep, there are plenty of others like that throughout the codebase, I'd just fix at least those in app (and mark others as deprecated for removal in QGIS 4?)

Reply:
[...]
> > I think that the pre-existing onAttributeSelectionChanged(); also didn't need to be public, it was added by.
> 
> Yep, there are plenty of others like that throughout the codebase, I'd just fix at least those in app (and mark others as deprecated for removal in QGIS 4?)

Sounds like a plan :+1: